core/src/variables.rs

```diff
@@ -1,28 +1,180 @@
+use crate::CoreError;
 use serde::{Deserialize, Serialize};
 use std::cmp::Ordering;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use std::hash::Hash;
 
-// This is, for now, a placeholder file for variables. The data types it contains is (some) of the
-// once it will contain when the variable store is actually implemented.
+#[derive(Debug, Clone, Default)]
+pub struct VariableStore(HashMap<String, Value>);
 
-// The type of a variable
+impl VariableStore {
+    pub fn get(&self, name: &str) -> Option<&Value> {
+        self.0.get(name)
+    }
+
+    /// Clears the variable store
+    pub fn clear(&mut self) {
+        self.0.clear()
+    }
+
+    /// Ensure that a valid name is used.
+    /// Only ASCII alphanumerics characters and "_" is allowed. The name may also not start with a number.
+    fn validate_name(&self, name: &str) -> Result<(), CoreError> {
```

Comment:
This issue doesn't seem to be resolved (not having it as an instance function, plus maybe renaming)